include/fiction/algorithms/simulation/sidb/detect_bdl_wires.hpp

```diff
@@ -85,18 +61,28 @@ struct detect_bdl_wires_params
  *
  * @tparam CellType Cell type.
  */
-template <typename CellType>
+template <typename Lyt>
 struct bdl_wire
 {
     /**
      * Vector of BDL pairs representing the wire.
      */
-    std::vector<bdl_pair<CellType>> pairs{};
+    std::vector<bdl_pair<cell<Lyt>>> pairs{};
 
     /**
      * Direction of the BDL wire.
      */
-    bdl_wire_direction direction{bdl_wire_direction::NO_DIRECTION};
+    port_direction direction{port_direction::NONE};
```

Comment:
I would rather rename this to `ports` because checking for `direction.dir` seems confusing.